dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -448,12 +447,50 @@ HRESULT WINAPI CDesktopFolder::CompareIDs(LPARAM lParam, PCUIDLIST_RELATIVE pidl
 {
     bool bIsDesktopFolder1, bIsDesktopFolder2;
 
+    if (LOWORD(lParam) >= DESKTOPSHELLVIEWCOLUMNS)
+    {
+        ERR("Got invalid column pointer %Ix\n", LOWORD(lParam));
+        return E_INVALIDARG;
+    }
+
     if (!pidl1 || !pidl2)
     {
         ERR("Got null pidl pointer (%Ix %p %p)!\n", lParam, pidl1, pidl2);
         return E_INVALIDARG;
     }
 
+    if (lParam == 3) // file modified date from index into DesktopSFHeader[]
```

Comment:
Please use an enum instead of magic constants (yes, I know the existing code does do it wrong, but at least the new additions can be written better).

Reply:
I'm happy to fix the correct column order and add enums if @Doug-Lyons does not want to, I just need to know if he plans to do it in this PR or not.

Reply:
I would like to leave adding the enum to @whindsaks in another PR for now. Thanks.

Reply:
I have worked with Whindmar to remove the magic constants (hopefully we found all of them).